### ISSUE TITLE
Restore-BcDatabaseFromArtifacts.ps1 needs SQL installed

### ISSUE THREAD

**Issue by kine** [microsoft/navcontainerhelper#1800]
Was trying the trick from Restore-BcDatabaseFromArtifacts to use admin module from artifact (outside container usage of the management module) but was not able because the libraries were not found. Module sqlps is part of SQL installation but I do not want to install that on my host. I was digging into it and have found that the libraries are part of sqlserver PS module. I slightly modified the code to load the libraries from there (preparing PR):
```
            import-module SqlServer
            $SqlModule = get-module SqlServer
            $Path = Split-Path $SqlModule.Path
            $Smo = [Reflection.Assembly]::LoadFile((Join-Path $Path 'Microsoft.SqlServer.Smo.dll'))
            $SmoExtended = [Reflection.Assembly]::LoadFile((Join-Path $Path 'Microsoft.SqlServer.SmoExtended.dll'))
            $ConnectionInfo = [Reflection.Assembly]::LoadFile((Join-Path $Path 'Microsoft.SqlServer.ConnectionInfo.dll'))
            $SqlEnum = [Reflection.Assembly]::LoadFile((Join-Path $Path 'Microsoft.SqlServer.SqlEnum.dll'))
                    
            $OnAssemblyResolve = [System.ResolveEventHandler] {
                param($sender, $e)
                if ($e.Name -like "Microsoft.SqlServer.Smo, Version=*, Culture=neutral, PublicKeyToken=89845dcd8080cc91") { return $Smo }
                if ($e.Name -like "Microsoft.SqlServer.SmoExtended, Version=*, Culture=neutral, PublicKeyToken=89845dcd8080cc91") { return $SmoExtended }
                if ($e.Name -like "Microsoft.SqlServer.ConnectionInfo, Version=*, Culture=neutral, PublicKeyToken=89845dcd8080cc91") { return $ConnectionInfo }
                if ($e.Name -like "Microsoft.SqlServer.SqlEnum, Version=*, Culture=neutral, PublicKeyToken=89845dcd8080cc91") { return $SqlEnum }
                foreach($a in [System.AppDomain]::CurrentDomain.GetAssemblies()) {
                    if ($a.FullName -eq $e.Name) { return $a }
                }
                return $null
            }
            [System.AppDomain]::CurrentDomain.add_AssemblyResolve($OnAssemblyResolve)
```

Now it is possible to use the cmdlets like Export-NAVApplication etc. outside container without SQL installed. Installed must be only SqlServer module.

**Comment by freddydk** [comment]
Do you get the smoServer as part of SqlServer as well?

**Comment by kine** [comment]
No, but you do not need it... or at least all worked for me witout using it.

**Comment by freddydk** [comment]
I have added some comments - the current PR breaks the scenario of running SQL Server on the host

**Comment by freddydk** [comment]
Shipped in 2.0.9
